### PR TITLE
drivers: sensor: Add suport for LTR329 sensor

### PR DESCRIPTION
**Pull request by konrad1s**
Hello
this pull request adds support the Liteon LTR329 Digital Ambient Light Sensor with I2C bus interface.

Datasheet for the sensor is available at the link: [datasheet](https://www.mouser.com/datasheet/2/239/Lite-On_LTR-329ALS-01%20DS_ver1.1-348647.pdf)
The ambient light conversion equation is as shown in the following document: [LTR-303ALS & LTR-329ALS Appendix A](https://cdck-file-uploads-europe1.s3.dualstack.eu-west-1.amazonaws.com/arduino/original/3X/8/d/8ddfc2d335f178e70e20974439a94ce31dd9df35.pdf) 
I have adapted the ambient light conversion equation to use fixed-point calculations instead of floating-point, enhancing efficiency.

### CODE REVIEW

File: drivers/sensor/ltr329/ltr329.c [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+		return (*output == 0) ? -EINVAL : 0;
+	}
+
+	return -EINVAL;
+}
+
+static int ltr329_get_mapped_int_time(const uint8_t register_val, uint8_t *const output)
+{
+	/* Map the register value to the integration time used in the lux calculation */
+	const int32_t int_time_lux_calc[] = {10, 5, 20, 40, 15, 25, 30, 35};
```
Comment (kartben) [inline drivers/sensor/ltr329/ltr329.c:277]:
```suggestion
	const uint8_t int_time_lux_calc[] = {10, 5, 20, 40, 15, 25, 30, 35};
```

File: dts/bindings/sensor/ltr,329.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+description: Lite-on LTR329 Digital Ambient Light Sensor
+
+compatible: "ltr,329"
+
+include: [sensor-device.yaml, i2c-device.yaml]
+
+properties:
+  gain:
+    type: int
+    default: 0
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ltr,329.yaml:13]: Need to justify in the description why this default value was chosen. Is it a register reset value?

File: dts/bindings/sensor/ltr,329.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      - 1 # Gain 2X (0.5 lux to 32k lux)
+      - 2 # Gain 4X (0.25 lux to 16k lux)
+      - 3 # Gain 8X (0.125 lux to 8k lux)
+      - 6 # Gain 48X (0.02 lux to 1.3k lux)
+      - 7 # Gain 96X (0.01 lux to 600 lux)
+    enum: [0, 1, 2, 3, 6, 7]
+
+  integration-time:
+    type: int
+    default: 0
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ltr,329.yaml:27]: Need to justify in the description why this default value was chosen. Is it a register reset value?

File: dts/bindings/sensor/ltr,329.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      - 3 # 400ms
+      - 4 # 150ms
+      - 5 # 250ms
+      - 6 # 300ms
+      - 7 # 350ms
+    enum: [0, 1, 2, 3, 4, 5, 6, 7]
+
+  measurement-rate:
+    type: int
+    default: 3
```
Comment (MaureenHelm) [inline dts/bindings/sensor/ltr,329.yaml:43]: Need to justify in the description why this default value was chosen. Is it a register reset value?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by konrad1s** [comment]
Hello, 
I fixed all review comments (changes added to the new commit).
I also changed the vendor from 'ltr' to 'liteon'. The previous prefix 'ltr' was specifically referring to the LiteOn Optical Sensor family [link](https://optoelectronics.liteon.com/en-global/led/LED-Component) which could lead to misunderstandings.

I intended to work on implementing an emulator, but I'm still grappling with some problems. The main one is lack of time. I will try to do this (if its not required, perhaps in the next PR)

File: dts/bindings/sensor/ltr,329.yaml (hunk 2 above)
Comment (konrad1s) [reply dts/bindings/sensor/ltr,329.yaml]: register reset value, comment added

File: dts/bindings/sensor/ltr,329.yaml (hunk 3 above)
Comment (konrad1s) [reply dts/bindings/sensor/ltr,329.yaml]: register reset value, comment added

File: dts/bindings/sensor/ltr,329.yaml (hunk 4 above)
Comment (konrad1s) [reply dts/bindings/sensor/ltr,329.yaml]: register reset value, comment added

File: drivers/sensor/ltr329/ltr329.c [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+	uint8_t status;
+	int rc;
+
+	do {
+		rc = ltr329_read_register(bus, LTR329_ALS_STATUS, &status);
+		if (rc < 0) {
+			LOG_ERR("Failed to get LTR329 ALS_STATUS");
+			return rc;
+		}
+		k_sleep(K_MSEC(3));
```
[the remark this answers is not in the document]
Comment (konrad1s) [reply drivers/sensor/ltr329/ltr329.c]: removed, return -EBUSY added (I saw a similar solution in other sensors)

File: dts/bindings/sensor/ltr,329.yaml [hunk 6]
```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2023 Konrad Sikora
+# SPDX-License-Identifier: Apache-2.0
+
+description: Lite-on LTR329 Digital Ambient Light Sensor
+
+compatible: "ltr,329"
```
[the remark this answers is not in the document]
Comment (konrad1s) [reply dts/bindings/sensor/ltr,329.yaml]: Updated, changed vendor from ltr to liteon

**Pull request closed without merging**